metron-platform/metron-writer/src/main/java/org/apache/metron/writer/hdfs/HdfsWriter.java

```diff
@@ -85,7 +85,17 @@ public void init(Map stormConfig, TopologyContext topologyContext, WriterConfigu
     this.fileNameFormat.prepare(stormConfig,topologyContext);
     if(syncPolicy != null) {
       //if the user has specified the sync policy, we don't want to override their wishes.
-      syncPolicyCreator = (source,config) -> syncPolicy;
+      syncPolicyCreator = (source,config) -> {
+        try {
```

Comment:
This seems a little black magic-y.  Can we expand the comment as to why we need to clone?